aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -22,14 +22,14 @@ void TensorIterator::reorder_dimensions() {
 
   // returns 1 if the dim0 should come after dim1, -1 if dim0 should come
   // before dim1, and 0 if the comparison is ambiguous.
-  auto should_swap = [&](int dim0, int dim1) {
+  auto should_swap = [&](size_t dim0, size_t dim1) {
```

Comment:
shouldn't these be `int64_t` because that's whats passed in?

Reply:
The situation here is just bad, because indexing of stride_bytes[dim0], etc. can only use size_t and technically not int64_t (although perhaps it practice doesn't matter if this isn't ran on 32-bit systems).

All of TensorIterator looks to not be typed properly, per our discussion. Perhaps we need dedicated typedef for dimension size type... so will leave this for later.